server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -1989,7 +1985,9 @@ private CompactionStats _majorCompact(MajorCompactionReason reason)
           CompactionEnv cenv = new CompactionEnv() {
             @Override
             public boolean isCompactionEnabled() {
-              return Tablet.this.isCompactionEnabled();
+              // avoid calling isClosing() because its synchronized and this is called frequently in
+              // compaction
+              return closeState != CloseState.CLOSING;
```

Comment:
The comment helps a little, but I'm still confused why synchronization isn't required here, but is for all the callers of `isClosing()` (which effectively has an identical implementation).

Reply:
I don't think sync is required for isClosing().   I thought about changing it but it is part of a family of methods for checking state, and if its changed then all should be changed.  One of the methods in that family reads closeState twice, so it would need a slight refactoring (read into a local var once) if it were not synchronized.  I mulled over making that change and decided not to.  I didn't want to make too much of a change for 1.9.  If I were to make a larger change would you be able to give it a thorough review?

Reply:
One thing  Iike about this change was that it does nott matter if isClosing was sync or not, its very explicit about what it requires.  If isClosing were made unsync a comment could be added to it saying that it should not be made sync.  I suspect in the past that isClosing was not a sync method, but I didn't do the research to confirm this.

Reply:
> If I were to make a larger change would you be able to give it a thorough review?

I think it's fine for 1.9. Could do follow-on in 2.0 to do a more thorough analysis of whether these need to be synchronized.

Reply:
I opened #1037 this would be a good task for 2.0.x